doc/topics/the-event-loop-timers-and-nexttick.md

```diff
@@ -0,0 +1,373 @@
+## `process.nextTick()` vs `setImmediate()`
+
+We have two calls that are similar as far as users are concerned, but their
+names are confusing.
+
+* `process.nextTick()` fires immediately on the same phase
+* `setImmediate()` fires on the following iteration or 'tick' of the event
+ loop
+
+In essence, the names should be swapped. `process.nextTick()` fires more
+immediately than `setImmediate()` but this is an artifact of the past which is
+unlikely to change. Making this switch would break a large percentage of the
+packages on npm. Every day more new modules are being added, which mean every
+day we wait, more potential breakages occur. While they are confusing, the
+names themselves won't change.
+
+*We recommend developers use `setImmediate()` in all cases because its easier
+to reason about (and it leads to code that's compatible with a wider
+variety of environments, like browser JS.)*
+
+## Why use `process.nextTick()`? 
+
+There are two main reasons:
+
+1. Allow users to handle errors, cleanup any then unneeded resources, or
+perhaps try the request again before the event loop continues.
+
+2. At times it's necessary to allow a callback to run after the call stack has
+unwound but before the event loop continues. 
+
+One example is to match the user's expectations. Simple example:
+
+```js
+var server = net.createServer();
+server.on('connection', function(conn) { });
+
+server.listen(8080);
+server.on('listening', function() { });
+```
+
+Say that listen() is run at the beginning of the event loop, but the listening
+callback is placed in a `setImmediate()`. Now, unless a hostname is passed 
+binding to the port will happen immediately. Now for the event loop to proceed
+it must hit the `poll` phase, which means there is a non-zero chance that a 
+connection could have been received allowing
+the connection event to be fired before the listening event.
+
+Another example is running a function constructor that was to, say, 
+inherit from `EventEmitter` and it wanted to call an event within the
+constructor:
+
+```js
+const EventEmitter = require('events');
+const util = require('util');
+
+function MyEmitter() {
+  EventEmitter.call(this);
+  this.emit('event');
+}
+util.inherits(MyEmitter, EventEmitter);
+
+const myEmitter = new MyEmitter();
+myEmitter.on('event', function() {
+  console.log('an event occurred!');
+});
+```
+
+You can't emit an event from the constructor immediately 
+because the script will not have processed to the point where the user 
+assigns a callback to that event. So, within the constructor itself, 
+you can use `process.nextTick()` to set a callback to emit the event 
+after the constructor has finished, which provides the expected results:
+
+```js
+const EventEmitter = require('events');
+const util = require('util');
+
+function MyEmitter() {
+  EventEmitter.call(this);
+
+  // use nextTick to emit the event once a handler is assigned
+  process.nextTick(() => {
+    this.emit('event');
+  }.bind(this));
```

Comment:
Thanks @evanlucas !  replaced the arrow function with a plain-old anonymous function.